components/21-atoms/table/table-mobile.functions.js

```diff
@@ -0,0 +1,159 @@
+'use strict';
+
+/* global define, module */
+(function (root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define(factory);
+  }
+  else {
+    if (typeof exports === 'object') {
+      module.exports = factory();
+    }
+    else {
+      root.MobileTable = factory();
+    }
+  }
+}(this || window, function () {
+  return function (element, options = {}) {
+
+    /**
+     * The table caption.
+     */
+    let caption = null;
+
+    /**
+     * The responsive list version of the table.
+     */
+    let tableList = null;
+
+    /**
+     * Override default options with options param.
+     */
+    options = (() => {
+      const defaults = {
+        headingType: 'h3'
+      };
+
+      const keys = Object.keys(defaults);
+      let newOptions = options || {};
+
+      for (let i = keys.length; i--;) {
+        newOptions[keys[i]] = options[keys[i]] || defaults[keys[i]];
+      }
+
+      return options;
+    })();
+
+    /**
+     * Add a wrapper div with class .table-list as next sibling to the table.
+     */
+    const addTableList = () => {
+      tableList = document.createElement('div');
+      tableList.classList.add('table-list');
+      tableList.setAttribute('aria-labelledby', 'list-description');
+      element.parentNode.insertBefore(tableList, null);
+
+      addList();
+    };
+
+    /**
+     * Adds the individual items to the list <ul></ul> of the table-list wrapper.
+     *
+     * @param {Object} list
+     *   The list to add in elements based on the table.
+     */
+    const addListItems = (list) => {
+      // Get the headers of the table columns.
+      let colHeadingsNodeList = element.querySelector('table').querySelectorAll('th[scope="col"');
+      // Get the headers of the table rows.
+      let rowHeadingsNodeList = element.querySelector('table').querySelectorAll('th[scope="row"]');
+      // Get all rows of the table.
+      let rowsNodeList = element.querySelector('table').querySelectorAll('tr');
+      // The first row of the table.
+      const firstRow = element.querySelector('tr');
+
+      // Determine if we have column headings.
+      options.columnHeaders = firstRow.getElementsByTagName('th').length === firstRow.querySelectorAll('*').length;
```

Comment:
This will fail.
```
<tr>
  <th>
    <span>trlala></span>
    <i class=icon></i>
  </th>
</tr>
```

Reply:
What if it has to be a table with only one column?... this could be a valid table no? @hstandaert what do you think?

Reply:
That's beside the point.
The number of 'th' tags inside a row does not have to equal the total number of elements inside that row. And in many cases, it won't (icons, sorting buttons, spans, layout elements...)

Reply:
```suggestion
      options.columnHeaders = firstRow.getElementsByTagName('th').length === firstRow.children.length;
```

My bad, guys 🙃 I was trying to use [Node.childNodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/childNodes) yesterday (which also returns textContent and therefore didn't return the expected result).

We should use [ParentNode.children](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/children).